packages/devtools-reps/src/reps/tests/element-node.js

```diff
@@ -0,0 +1,320 @@
+  });
+
+  it("renders and inspect icon", () => {
+    const onInspectIconClick = jest.fn();
+
+    const renderedComponent = shallow(ElementNode.rep({
+      object: stub,
+      onInspectIconClick
+    }));
+
+    const node = renderedComponent.find(".open-inspector");
+    node.simulate("click", { type: "click" });
+
+    expect(onInspectIconClick.mock.calls.length).toEqual(1);
+    expect(onInspectIconClick.mock.calls[0][0]).toEqual(stub);
+    expect(onInspectIconClick.mock.calls[0][1].type).toEqual("click");
+  });
+
+  it("calls the expected argument when mouseout is fired on Rep", () => {
+    const onDOMNodeMouseOut = jest.fn();
+    const renderedComponent = shallow(ElementNode.rep({
+      object: stub,
+      onDOMNodeMouseOut
+    }));
+
+    renderedComponent.simulate("mouseout");
+
+    expect(onDOMNodeMouseOut.mock.calls.length).toEqual(1);
+  });
+
+  it("calls the expected argument when mouseover is fired on Rep", () => {
+    const onDOMNodeMouseOver = jest.fn();
+    const renderedComponent = shallow(ElementNode.rep({
+      object: stub,
+      onDOMNodeMouseOver
+    }));
+
+    renderedComponent.simulate("mouseover");
+
+    expect(onDOMNodeMouseOver.mock.calls.length).toEqual(1);
+  });
+});
+
+describe("ElementNode - Node with leading and trailing spaces class name", () => {
+  const stub = stubs.get("NodeWithLeadingAndTrailingSpacesClassName");
+
+  it("selects ElementNode Rep", () => {
+    expect(getRep(stub)).toBe(ElementNode.rep);
+  });
+
+  it("renders with expected text content", () => {
+    const renderedComponent = shallow(ElementNode.rep({
+      object: stub
+    }));
+
+    expect(renderedComponent.text())
+      .toEqual('<body id=\"nightly-whatsnew\" class=\" html-ltr    \">');
+  });
+
+  it("renders with expected text content in tiny mode", () => {
+    const renderedComponent = shallow(ElementNode.rep({
+      object: stub,
+      mode: MODE.TINY
+    }));
+
+    expect(renderedComponent.text())
+      .toEqual("body#nightly-whatsnew.html-ltr");
+  });
+});
+
+describe("ElementNode - Node without attributes", () => {
+  const stub = stubs.get("NodeWithoutAttributes");
+
+  it("selects ElementNode Rep", () => {
+    expect(getRep(stub)).toBe(ElementNode.rep);
+  });
+
+  it("renders with expected text content in tiny mode", () => {
+    const renderedComponent = shallow(ElementNode.rep({
+      object: stub,
+      mode: MODE.TINY
+    }));
+
+    expect(renderedComponent.text()).toEqual("p");
+  });
+});
+
+describe("ElementNode - Node with many attributes", () => {
+  const stub = stubs.get("LotsOfAttributes");
+
+  it("selects ElementNode Rep", () => {
+    expect(getRep(stub)).toBe(ElementNode.rep);
+  });
+
+  it("renders with expected text content", () => {
+    const renderedComponent = shallow(ElementNode.rep({
+      object: stub
+    }));
+
+    expect(renderedComponent.text()).toEqual(
+      '<p id="lots-of-attributes" n="" b="" c="" d="" e="" f="" a="" h="" ' +
```

Comment:
As discussed, let's revert this as it was in the mochitest, and fix the ordering in ElementNode rep